src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/IconTintColor/IconTintColorBehavior.windows.cs

```diff
@@ -130,7 +130,14 @@ void ApplyTintColor(FrameworkElement platformView, View element, Color? color)
 
 	void LoadAndApplyImageTintColor(View element, WImage image, Color color)
 	{
-		if (element is IImageElement { Source: UriImageSource uriImageSource })
+		// Sometimes WImage source doesn't match View source so the image is not ready to be tinted, have to wait for next ImageOpened event
+		if (element is IImageElement { Source: FileImageSource fileImageSource } &&
+			image.Source is BitmapImage bitmapImage &&
```

Comment:
This maybe a dumb question, but does the `BitmapImage` type on Windows support all image types, eg PNG, JPEG, etc?

Reply:
Yes it does https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.media.imaging.bitmapimage?view=windows-app-sdk-1.5

The IconTintColorBehavior is also using BitmapImage to get the image uri in the TryGetSourceImageUri() function, which is why I used it here too